bitrepository-webclient/src/main/webapp/table-modal.js

```diff
@@ -0,0 +1,171 @@
+                html += `<button class="next-button">Next</button>`;
+            }
+            // Show [current / total] pages.
+            html += `<p class="current-page-p">Page ${page} / ${maxPages}</p>`;
+            html += `</div>`;
+            html += `</div>`;
+
+            // Create table
+            html += `<table class="modal-table" style="width: 100%; border-collapse: separate;">`;
+
+            // Populate the header of the table.
+            let header = `<thead class="modal-table-head">`;
+            header += `<tr style="padding: 5px; border-bottom: 1pt solid black;">`;
+            header += `<th style="text-align: left; border-left: 1px solid white;">#</th>`;
+            header += `<th style="text-align: left; padding-left: 5px; width: 35%">File ID</th>`;
+
+            // Color header yellow for the pillar that is in 'focus'
+            for (let i = 0; i < pillars.length; i++) {
+                if (pillars[i] === pillarID) {
+                    header += `<th style="background-color: #ecf275;">${pillars[i].toUpperCase()}</th>`;
+                } else {
+                    header += `<th>${pillars[i].toUpperCase()}</th>`;
+                }
+            }
+
+            header += `</tr>`;
+            header += `</thead>`;
+            html += header + `<tbody id="files-table">`
+
+            // Populate the file status rows of the table. This only happens if there are more than 0 files.
+            let idxIncrement = (page - 1) * pageSize;
+            html += getTableBody(operation, json, idxIncrement);
+
+            html += `</tbody>`;
+            html += `</table>`;
+            html += `<p class="no-result-p" style="text-align:center; margin-top: 15px; display: none;">No results found</p>`;
+            html += `</div>`;
+
+            // Assign html and activate searchbar filtering and copy to clipboard.
+            $(contentElement).html(html);
+            activateSearchbar();
+            enableCopyToClipboard();
+
+            // Enable button functionality
+            if (page > 1) {
+                $(".prev-button").on("click", () => self.getModal(page - 1));
+            }
+            if (page < maxPages) {
+                $(".next-button").on("click", () => self.getModal(page + 1));
+            }
+        }).fail(function () {
+            let html = "<div class=\"alert alert-error\">"
+            html += "Failed to load page";
+            html += "</div>"
+            $(contentElement).html(html);
+        });
+    };
+
+    function getTableBody(operation, json, startIdx) {
+        let html = ``;
+
+        for (let i = 0; i < files.length; i++) {
+            html += `<tr style="border-top: 1px solid #9996">`;
+            html += `<td style="border-right: 1px solid #9996;">${startIdx + i + 1}</td>`;
+            html += `<td style="padding-left: 5px;"><a href="javascript:void(0);" class="file-id">${files[i]}</a></td>`;
+            for (let k = 0; k < pillars.length; k++) {
```

Comment:
I'd probably rename `k` to `j` for conventions sake as variable name `j` is no longer occupied with the change to the better name `json`. :)